### PR TITLE
Convert `setup.py` to `pyproject.toml`

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/egeakman/python-docs-bootstrapper/issues/4.

I used two tools for this:

* https://github.com/gvalkov/setuptools-py2cfg
* https://github.com/abravalheri/ini2toml

I used the first to convert `setup.py` to `setup.cfg`, and the second from `setup.cfg` to `pyproject.toml`, something like this:

```sh
pip install setuptools-py2cfg ini2toml
setuptools-py2cfg setup.py > setup.cfg
ini2toml setup.cfg > pyproject.toml
rm setup.*
```

Next, I added a few more pre-commit hooks, one to format `pyproject.toml` to make it easy to compare against other projects, and some others to check TOML and YAML files.

Then to check the output is okay, I ran:

```sh
pip install build
python -m build
```

Which creates an sdist and wheel in the `dist` directory. I compared the output from `master` and from this branch, and there were no unexpected differences.

We're still using setuptools here as the "build backend", which is just fine. We can also easily change it to others too. I tend to use hatchling, but it doesn't make too much difference.

### CODE REVIEW

**Comment by egeakman** [comment]
Thanks Hugo!

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Topic :: Utilities",
+]
+dependencies = [
+  "sphinx==4.5",
+]
+[project.urls]
+Download = "https://github.com/egeakman/python-docs-bootstrapper/archive/0.1.1.tar.gz"
```
Comment (egeakman) [inline pyproject.toml:42]: Is this supported?

```suggestion
Download = "https://github.com/egeakman/python-docs-bootstrapper/archive/{project.version}.tar.gz"
```

Comment (egeakman) [reply pyproject.toml:42]: Tried it, it is not. Is there a workaround or an alternative for variable interpolation?

Comment (hugovk) [reply pyproject.toml:42]: No, it leaves it as `{project.version}.tar.gz"`

Actually, I never use "Download" here, the usual thing is for people to install via PyPI.

Perhaps something like this instead?

```suggestion
Releases = "https://github.com/egeakman/python-docs-bootstrapper/releases"
```

Comment (egeakman) [reply pyproject.toml:42]: Makes sense, thanks!

**Review by egeakman — APPROVED** [review]

**Pull request merged**